scm/src/scm_input.F90

```diff
@@ -1006,293 +1159,602 @@ subroutine get_case_init_DEPHY(scm_state, scm_input)
+  integer :: input_n_init_times, input_n_forcing_times, input_n_lev, input_n_snow, input_n_ice, input_n_soil
   
   missing_value_eps = missing_value + 0.01
   
   !> - Open the case input file found in the processed_case_input dir corresponding to the experiment name.
-  call check(NF90_OPEN(trim(adjustl(scm_state%case_name))//'_SCM_driver.nc',nf90_nowrite,ncid))
+  call check(NF90_OPEN(trim(adjustl(scm_state%case_name))//'_SCM_driver.nc',nf90_nowrite,ncid),"nf90_open()")
   
   !> - Get the dimensions.
   
   !required dimensions
-  call check(NF90_INQ_DIMID(ncid,"t0",varID))
-  call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_init_times))
-  call check(NF90_INQ_DIMID(ncid,"time",varID))
-  call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_forcing_times))
-  call check(NF90_INQ_DIMID(ncid,"lev",varID))
-  call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_lev))
+  call check(NF90_INQ_DIMID(ncid,"t0",varID),"nf90_inq_dimid(t0)")
+  call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_init_times),"nf90_inq_dim(t0)")
+  call check(NF90_INQ_DIMID(ncid,"time",varID),"nf90_inq_dimid(time)")
+  call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_forcing_times),"nf90_inq_dim(time)")
+  call check(NF90_INQ_DIMID(ncid,"lev",varID),"nf90_inq_dimid(lev)")
+  call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_lev),"nf90_inq_dim(lev)")
   !Check whether long_name = 'altitude', units='m' OR long_name = 'pressure', units='Pa'?
   !It may not matter, because 'lev' may not be needed when the IC pressure and height are BOTH already provided
-  call check(NF90_INQ_DIMID(ncid,"lat",varID))
-  call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_lat))
-  call check(NF90_INQ_DIMID(ncid,"lon",varID))
-  call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_lon))
-  
+
   !### TO BE USED IF DEPHY-SCM can be extended to include model ICs ###
   !possible dimensions (if using model ICs)
   ierr = NF90_INQ_DIMID(ncid,"nsoil",varID)
   if(ierr /= NF90_NOERR) then
     input_n_soil = missing_soil_layers
   else
-    call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_soil))
+    call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_soil),"nf90_inq_dim(nsoil)")
   end if
   ierr = NF90_INQ_DIMID(ncid,"nsnow",varID)
   if(ierr /= NF90_NOERR) then
     input_n_snow = missing_snow_layers
   else
-    call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_snow))
+    call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_snow),"nf90_inq_dim(nsnow)")
   end if
   ierr = NF90_INQ_DIMID(ncid,"nice",varID)
   if(ierr /= NF90_NOERR) then
     input_n_ice = missing_ice_layers
   else
-    call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_ice))
+    call check(NF90_INQUIRE_DIMENSION(ncid, varID, tmpName, input_n_ice),"nf90_inq_dim(nice)")
   end if  
   
+  model_ics=.false.
```

Comment:
Does this need to stay? I thought that the UFS forcing generator script was outputting in DEPHY format, and thus reading in this file, you might have the model "restart"-type data. Isn't this disabling that?

Reply:
No. Good catch. Removed. I replaced this by querying the DEPHY global attribute surface_forcing_lsm.